### ISSUE TITLE
Add visualizations for macrostate composition

### ISSUE THREAD

**Issue by Marius1311** [theislab/cellrank#635]
- [ ] Additional function parameters, change functionality or change defaults?
- [ ] New estimator in `cr.tl.estimators`?
- [ ] New kernel in `cr.tl.kernels`?
- [ ] New gene trend model in `cr.ul.models`?
- [ ] New plotting function in `cr.pl`?
- [ ] Other?

After having computed macrostates, it's often very insightful to check which cells they're made up from, i.e. what the identity of the top `N` (by default, 30) cells is that have been assigned to any macrostate. For example, one may be interested in the distribution over clusters or time points (for time-series data). For an example plot, see Fig. 1 below, which shows the distribution over time-points. 

We should implement this as an estimator-level plotting method called `plot_macrostate_composition`.  Code for the actual plotting is given below. 

## Fig. 1
<img width="556" alt="Screenshot 2021-06-28 at 11 48 14" src="https://user-images.githubusercontent.com/24494650/123616600-bac7ed00-d806-11eb-8d49-5a75a4ec24a9.png">
Distribution over days. Note that the colors here aren't ideal. 

## Code
```python
def stacked_histogram(adata: AnnData, 
                      states_key: str = 'macrostates', 
                      cat_key: str ='cluster',
                      width: float = 0.8, 
                      figsize: Iterable[float] = (5, 4),
                      show_title: bool = True,
                      rotation: int = 0,
                      save: Optional[str] = None):
    """Plot stacked histogram for macrostats over categorical annotations
    
    Parameters
    ---
    adata
        Annotated data matrix
    states_key
        Key from `.obs` where macrostate clustering can be found
    cat_key
        Key from `.obs` where the categorical annotations can be found
    width
        Bar width
    figsize
        Size of the overall figure
    show_title
        Whether to display generated titles
    rotation
        x-label rotation
    save
        Path where the figure is to be saved
        
    Returns
    ---
    Noting, just plots
    """
    
    # pull annotations from AnnData object
    mask = ~adata.obs[states_key].isna()
    base_df = sc.get.obs_df(adata[mask], [cat_key, states_key])
    
    # define parameters
    macrostates = list(base_df[states_key].cat.categories)
    cats = list(base_df[cat_key].cat.categories)
    cats_colors = adata.uns[f"{cat_key}_colors"]
    ind = np.arange(len(macrostates))
    y_max = np.array([np.sum(base_df[states_key] == ms) for ms in macrostates]).max()
    
    # initialise the figure
    fig = plt.figure(None, figsize)
    bottom = np.array([0] * len(macrostates))

    # iterate over categories, plot one barchart per category
    for i, cat in enumerate(cats):
        frequencies = [np.sum(base_df.loc[base_df[states_key] == ms, cat_key] == cat) for ms in macrostates]
        plt.bar(ind, frequencies, width, label=cat, color=cats_colors[i], bottom=bottom)
        bottom += np.array(frequencies)
    
    # prettify the plot
    plt.ylabel('frequency')
    plt.xlabel('macrostate')
    plt.xticks(ind, macrostates)
    plt.xticks(rotation=rotation)
    
    if show_title:
        plt.title(f"distribution over `{cat_key}`")
    plt.ylim([0, y_max + 2])
    plt.legend(bbox_to_anchor=(1.05, 1), loc='upper left')
    
    if save is not None:
        plt.savefig(save)
    plt.show()
```

Note that I had to write annotations to AnnData before calling this funciton using
```
adata.obs['macrostates'] = g.macrostates
adata.uns['macrostates_colors'] = g.macrostates_memberships.colors
```

**Comment by Marius1311** [comment]
Note: there was a bug in my original implementation above, colors in the stacked bar chart did not match the correct colors corresponding to the categorical annotations, see Fig. 1 & 2 below. For example, day 12 should be red-ish, but it's black. 

## Correct categorical annotations
![image](https://user-images.githubusercontent.com/24494650/123660036-97676700-d833-11eb-9fc1-af6a59624b09.png)


## Stacked bar chart
![image](https://user-images.githubusercontent.com/24494650/123659976-89194b00-d833-11eb-86bc-595bb2a1d540.png)

**Comment by michalk8** [comment]
> Note: there was a bug in my original implementation above, colors in the stacked bar chart did not match the correct colors corresponding to the categorical annotations, see Fig. 1 & 2 below. For example, day 12.5 should be red-ish, but it's black.

Ok, is it fixed in the snippet above?

**Comment by Marius1311** [comment]
No, the bug sits very much in there.

**Comment by michalk8** [comment]
closed via #641
